clamav/cnudie.py

```diff
@@ -14,29 +14,122 @@
 import clamav.scan
 import cnudie.iter
 import dso.model
+import delivery.model
+import delivery.client
 import github.compliance.model
 import oci.client
 
 logger = logging.getLogger(__name__)
 ci.log.configure_default_logging()
 
 
+def _finding_for_component_artifact(
+    previous_findings: list[delivery.model.ArtefactMetadata],
```

Comment:
that being said, I do not like the pattern of "return first element or return None" like that.

consider any of:
- itertools (did not check whether they have sth.)
- for-loop + return
- next + try-except (StopIteration)

reasons:
- KISS
- readability
- your pattern does not actually save a lot of lines of code (not that this actually would be a big concern)